packages/cardpay-cli/builder.ts

```diff
@@ -9,31 +9,44 @@ export function buildYargs(args: string[]) {
     .help()
     .command(commands)
     .options({
+      connectionType: {
+        alias: 'c',
+        type: 'string',
+        description: 'Connection type',
+        demandOption: true,
+      },
+      network: {
+        type: 'string',
+        description: 'Network',
+        demandOption: true,
+      },
       mnemonic: {
         alias: 'm',
         default: process.env.MNEMONIC_PHRASE,
         type: 'string',
-        description: 'Phrase for mnemonic wallet. Also can be pulled from env using MNEMONIC_PHRASE',
-      },
-      walletConnect: {
-        alias: 'w',
-        type: 'boolean',
-        description: 'A flag to indicate that wallet connect should be used for the wallet',
+        description: 'Phrase for mnemonic wallet',
       },
     })
     .check((argv) => {
       if (process.env.BUILDING_README) {
         return true;
       }
-
-      if (!argv.mnemonic && !argv.walletConnect) {
-        return 'Wallet is not specified. Either specify that wallet connect should be used for the wallet, or specify the mnemonic as a positional arg, or pass the mnemonic in using the MNEMONIC_PHRASE env var';
-      }
-      // is this the right way to do this?
-      if (argv.walletConnect) {
-        argv.mnemonic = undefined;
+      switch (argv.connectionType) {
+        case 'mnemonic':
+          if (!argv.mnemonic) {
+            return `Must specify '--mnemonic' when using connectionType of 'mnemonic'`;
+          } else {
+            argv.web3Opts = {
+              mnemonic: argv.mnemonic,
+            };
+            return true;
+          }
+        case 'wallet-connect':
+          return true;
+        case 'trezor':
+          return true;
+        default:
+          return `Must choose either 'mnemonic, wallet-connect, treozr' for connectionType`;
       }
-      return true;
-    })
-    .demandOption(['network'], `'network' must be specified.`);
+    });
```

Comment:
@habdelra 

Unfortunately, I have made the commands more verbose

**old**
```
yarn cardpay safe list --network sokol --mnemonic "hi bye"
yarn cardpay safe list --network sokol --walletConnect
```

**new**
```
yarn cardpay safe list --network sokol -c mnemonic --mnemonic "hi bye"
yarn cardpay safe list --network sokol -c wallet-connect
```

I can make commands backward compatible but im not sure if we want to do that

Reply:
@lukemelia comments on this?

Reply:
i don't see why there has to be a direct correlation between the arguments and the connection type that `getOpts` uses. why can't we still retain the original commands and add `--trezor`, and then in `getOpts` sort build the `Web3Opts` object?

Reply:
Ok @habdelra I have made the change to preserve the shape of the commands.